### ISSUE TITLE
Expose VSCode Remote API

### ISSUE THREAD

**Issue by robertripoll** [microsoft/vscode-remote-release#10400]
Type: Feature Request

---

I am developing a VSCode extension that loads passwords from an external service. I would like to use these passwords whenever the VSCode Remote extension prompts for a password, and therefore autofill it for the user. It would be useful if the VSCode Remote extension’s API were made available to other extensions, allowing them to access and use passwords in a similar way.

**Comment by joshspicer** [comment]
Remote extensions do not store your password.  Do you have a more specific example from one of the remote extensions?  If so, perhaps I can provide a more specific pointer  From what I understand here, you should have all the tools you need to build your extension!

**Comment by robertripoll** [comment]
I didn't explain myself properly. Let me give it another try :)

I'm currently building a VSCode extension to autofill KeePass passwords within the IDE on behalf of the user. I found a case when the VSCode Remote SSH extension could prompt the user for an SSH password when logging into a remote server or computer. I thought about triggering this connection manually from within my extension (for example, calling an API) so that I would also be able to autofill the SSH password automatically. Would it require VSCode Remote Extension to expose its API to other extensions (such as mine)? Or is there another better way to do this?

Thanks!

**Comment by robertripoll** [comment]
Bump

**Comment by joshspicer** [comment]
Are you connecting to a remote machine to do this keepass autofilling, or are you interested in the ability to prompt for a password and write it into a terminal?

This latter functionality can be accomplished with public VS Code APIs.  For this example you can use the QuickPick API and the Terminal API to accept user input and fill it into a terminal.

**Comment by robertripoll** [comment]
@joshspicer nope, I’m actually using VSCode locally to connect to a remote server via the VSCode Remote SSH extension. The remote server is configured to use a password instead of an SSH certificate. VSCode prompts for a password for this remote, but my idea is to autofill it using another VSCode extension (one that I plan to create). This extension would fill passwords from a KeePass database.

However, the problem is that there’s currently no way to achieve this. The popup dialog for entering the password doesn’t allow it. I considered calling an internal API of the VSCode Remote SSH extension to pass the password directly, so the user wouldn’t need to enter it manually. Unfortunately, the VSCode Remote SSH extension doesn’t expose any “public” API for other to extensions to call.
